test/TestContour.cc

```diff
@@ -11,27 +11,23 @@
 #include "Util/Message.h"
 #include "src/Frame/Frame.h"
 
-static const std::string IMAGE_OPTS = "-s 0";
-static const std::string IMAGE_OPTS_NAN = "-s 0 -n row column -d 10";
-
 class ContourTest : public ::testing::Test {
 public:
-    void GenerateContour(
-        int width, int height, std::string image_opts, const CARTA::FileType& file_type, const CARTA::SmoothingMode& smoothing_mode) {
-        std::string image_shape = std::to_string(width) + " " + std::to_string(height);
+    void GenerateContour(std::string filename, const CARTA::FileType& file_type, const CARTA::SmoothingMode& smoothing_mode) {
         std::string file_path;
         if (file_type == CARTA::FileType::HDF5) {
-            file_path = ImageGenerator::GeneratedHdf5ImagePath(image_shape, image_opts);
+            file_path = (Hdf5Images() / filename);
         } else {
-            file_path = ImageGenerator::GeneratedFitsImagePath(image_shape, image_opts);
+            file_path = (FitsImages() / filename);
         }
```

Comment:
This conditional should be removed entirely: you should construct the paths outside this function, and pass in the constructed path, not the filename and type.